packages/custom_lint/test/cli_test.dart

```diff
@@ -21,32 +21,87 @@ final helloWordPluginSource = createPluginSource([
   )
 ]);
 
-void main() {
-  test('exits with 0 when no lint and no error are found', () async {
-    final plugin = createPlugin(name: 'test_lint', main: emptyPluginSource);
+String progressMessage({required bool supportsAnsiEscapes}) {
+  const analyzing = 'Analyzing...';
+  if (supportsAnsiEscapes) {
+    return '$analyzing                            \b/\b0.0s';
```

Comment:
I don't think we should test this, the output is not deterministic, it may sometimes take longer. And it is not working on the Github runner. The terminal there doesn't seem to support it so I can't even force enable it in the test.

Reply:
Tests should be independent from the platform environment. It shouldn't matter whether the CI supports ansi or not. We should still be able to test both the Ansi and non-Ansi scenarios.

Reply:
We do want to test this IMO

We shouldn't write a test saying "this takes exactly x.ys". But rather we should have a test saying "we start a timer when the command starts, and the timer stops when the analysis completes".

For example one particular case would be the --watch mode. We want to make sure that the progress indicator stops at the right moment.